### ISSUE TITLE
Operator does not react when a new ServiceAccount is created

### ISSUE THREAD

**Issue by RothAndrew** [alexellis/registry-creds#24]
**Describe the bug**
It looks like the operator does not react when a new ServiceAccount is created after the operator has been installed.

**To Reproduce**

1. Create new kind cluster
1. Follow instructions to install registry-creds, create seed secret, and add the ClusterPullSecret CRD
1. create a private image on dockerhub by pulling `nginx:latest`, retag, and push to your private dockerhub repo
1. Create a new basic helm chart with `helm create foobar` - which creates by default a new ServiceAccount and deployment of `library/nginx`
1. Change the repository and tag in values.yaml to your new private image
1. Run `helm install foobar ./foobar`
    1. A ServiceAccount is created in the `default` namespace, that the registry-creds operator does not react to and add the imagePullSecret
    1. A deployment is created that fails with ImagePullBackoff
1. Delete the registry-creds pod (and let it be recreated)
1. foobar deployment successfully pulls, since new registry-creds pod now sees the new ServiceAccount

Full recreation:
```sh
mkdir recreate-issue

cd recreate-issue

kind create cluster

kubectl apply -f https://raw.githubusercontent.com/alexellis/registry-creds/master/manifest.yaml

kubectl create secret docker-registry registry-creds-secret --namespace registry-creds-system --docker-username=rothandrew --docker-password=[REDACTED] --docker-email=roth.andy@gmail.com

cat <<EOF | kubectl apply -f -
apiVersion: ops.alexellis.io/v1
kind: ClusterPullSecret
metadata:
  name: docker-registry
spec:
  secretRef:
    name: registry-creds-secret
    namespace: registry-creds-system
EOF

helm create foobar

helm install foobar ./foobar --set image.repository=rothandrew/testprivate --set image.tag=test1

# foobar deployment is now in ErrImagePull with following error:
# Failed to pull image "rothandrew/testprivate:test1": rpc error: code = Unknown desc = failed to pull and unpack image "docker.io/rothandrew/testprivate:test1": failed to resolve reference "docker.io/rothandrew/testprivate:test1": pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed

kubectl get pods -n registry-creds-system --no-headers=true | awk '/registry-creds-registry-creds-controller/{print $1}' | xargs kubectl delete -n registry-creds-system pod

# foobar pod will now get pulled successfully the next time k8s tries.
```

**Expected behavior**
The operator notices when new service accounts are created and adds the ImagePullSecret to them

**Screenshots or console output**
N/A

**Operating system and version:**
MacOS Mojave 10.14.6

**List all possible solutions, and your suggested option**
Not sure. I'm not familiar with this operator's functionality enough to make any educated guesses.

**Additional context**
none

**Comment by alexellis** [comment]
@RothAndrew whilst I appreciate the instructions, these need to be steps we can actually run (which they are not). Hopefully it won't take you long to do that and create a little repo.

Please also see the full report and all the fields required.

https://raw.githubusercontent.com/alexellis/registry-creds/master/.github/ISSUE_TEMPLATE/bug_report.md

**Comment by RothAndrew** [comment]
@alexellis I will update the issue to meet your criteria

**Comment by alexellis** [comment]
All we should need is you to install the chart.

Then create a ServiceAccount in a namespace. That is what you're saying causes an issue right? What's the need for helm and all other other manual steps?

**Comment by alexellis** [comment]
@developer-guy did you want to try out the scenario above? From what I read in the code the operator should handle this use-case already.

**Comment by RothAndrew** [comment]
Just coming up with a full recreation of what I experienced. I'm sure there are lots of different ways this can be tested.

I'm working on coming up with a list of actual commands you can run

**Comment by RothAndrew** [comment]
Full list of commands to recreate are added. I will add the other required items now

**Comment by RothAndrew** [comment]
@alexellis Issue description should now fully comply

**Comment by developer-guy** [comment]
Thank you @RothAndrew for the explanatory description, I'll investigate this issue soon .

**Comment by RothAndrew** [comment]
More context: cloning the repo and running `make install && make run` (rather than running `kubectl apply -f https://raw.githubusercontent.com/alexellis/registry-creds/master/manifest.yaml`) does not exhibit the behavior I detailed above. It looks to be working correctly.

I will keep investigating

**Comment by RothAndrew** [comment]
Bumping the tag of `ghcr.io/alexellis/registry-creds` on line 232 of `manifest.yaml` to `0.2.6` fixes the issue for me. I'd be happy to submit the one line change as a PR if you'd like.

**Comment by developer-guy** [comment]
> Bumping the tag of `ghcr.io/alexellis/registry-creds` on line 232 of `manifest.yaml` to `0.2.6` fixes the issue for me. I'd be happy to submit the one line change as a PR if you'd like.

Yeah, I think that's true,  it fixed this issue for me too, I think the tag 0.25.0 does not include the feature that watches service account creation. 🤔

**Comment by alexellis** [comment]
/lock: inactive, raise your own issue as a GitHub Sponsor
